File: Find the Duplicate Number/Find_the_Duplicate_Number.java
class Solution {
    public int findDuplicate(int[] nums) {
        HashMap<Integer, Integer> mpp = new HashMap<>();
        for (int i : nums) {
            mpp.put(i, mpp.getOrDefault(i, 0) + 1);
        }
        for (Map.Entry<Integer, Integer> entry : mpp.entrySet()) {
            if (entry.getValue() >= 2) {
                return entry.getKey();
            }
        }
        return -1;
    }
}

File: Find the Duplicate Number/Find_the_Duplicate_Number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        mpp = {}
        for i in nums:
            mpp[i] = mpp.get(i, 0) + 1
        for k, v in mpp.items():
            if v >= 2:
                return k
        return -1
        